index.d.ts

```diff
@@ -86,7 +86,29 @@ declare namespace winston {
     exceptionHandlers?: any;
   }
 
-  interface Logger extends NodeJSStream.Transform {
+  type DefaulLevels = {
+    // for cli and npm levels
+    error: number;
+    warn: number;
+    help: number;
+    data: number;
+    info: number;
+    debug: number;
+    prompt: number;
+    http: number;
+    verbose: number;
+    input: number;
+    silly: number;
+    
+    // for syslog levels only
+    emerg: number;
+    alert: number;
+    crit: number;
+    warning: number;
+    notice: number;
+  }
+
+  type Logger<T extends Config.AbstractConfigSetLevels = DefaulLevels> = NodeJSStream.Transform & {
```

Comment:
To prevent any breaking changes, generic should be `T extends Config.AbstractConfigSetLevels = DefaultLevels | Config.AbstractConfigSetLevels`